### PR TITLE
OC-28430: fix library search box missing in Study Designer

### PR DESCRIPTION
**Pull request by umangzala-oc**
### 🗒️ Checklist

1. [ ] run linter locally
2. [ ] update developer docs (API, README, inline, etc.), if any
3. [ ] for user-facing doc changes create a Zulip thread at `#Support Docs Updates`, if any
4. [ ] draft PR with a title `<type>(<scope>)<!>: <title> DEV-1234`
5. [ ] assign yourself, tag PR: at least `Front end` and/or `Back end` or `workflow`
6. [ ] fill in the template below and delete template comments
7. [ ] review thyself: read the diff and repro the preview as written
8. [ ] open PR & confirm that CI passes & request reviewers, if needed
9. [ ] delete this section before merging

### 📣 Summary
Bug fix for missing search box in Library in Study Designer (dev). Root cause: OC-27826 upgrade replaced header.es6 (used startsWith) with mainHeader.component.tsx (uses strict ===). Dev appends ?econsent=ACTIVE to the library URL hash, which caused the route check to fail, hiding the search box. Fix strips the query string from the hash path before comparison in getCurrentPath() (routerUtils.ts).

1. ℹ️ have an account and a project
2. do this
3. do that
4. 🔴 [on main] notice that this isn't anywhere
5. 🟢 [on PR] notice that this is here
6. do that another thing
7. 🟢 notice that this changed like that

### CODE REVIEW

File: jsapp/js/router/routerUtils.ts [hunk 1]
```diff
@@ -36,7 +36,9 @@ export function redirectToLogin() {
 
 export function getCurrentPath(): string {
   const route = location.hash.split('#')
-  return route.length > 1 ? route[1] : ''
+  // Strip query string so path comparisons (isMyLibraryRoute etc.) work correctly
+  // even when a query param (e.g. ?econsent=ACTIVE) is present in the hash.
+  return route.length > 1 ? route[1].split('?')[0] : ''
```
[the remark this answers is not in the document]
Comment (umangzala-oc) [reply jsapp/js/router/routerUtils.ts]: Valid point but out of scope here — Library is only opened from Study Designer where user is already logged in, so getLoginUrl() is never triggered in this flow.

Comment (gushil) [inline jsapp/js/router/routerUtils.ts]: `getCurrentPath()` is shared by ~28 callers, including `getLoginUrl()` (line 26), which encodes its return value straight into the `next=` login-redirect param. Stripping the query string here silently changes that contract too.

Repro: user on `#/library/asset/new?econsent=ACTIVE` hits a session expiry -> `RequireAuth.tsx` -> `redirectToLogin()` -> `getLoginUrl()` now builds `next=/#/library/asset/new` with `econsent` dropped. `oc_tenant_auth/views.py` forwards `next` verbatim through OIDC, so the user lands back without the query param. `econsentSignature.ts`'s `getStudyEConsentModuleStatus()`/`getFormEventType()` read `window.location.hash` directly, get `null`, and `isEConsentAllowedEventType()` treats missing `event_type` as "no context" -> returns `true` — so this doesn't just hide the eConsent Signature item type, it can incorrectly re-enable it outside its allowed event context.

Suggest scoping the strip to the actual bug site instead of this primitive: add a separate helper (e.g. `getCurrentRoutePath()`) for `isMyLibraryRoute()`/`isPublicCollectionsRoute()`, and leave `getCurrentPath()` (and `getLoginUrl()`) returning the full hash including query.

**Review by gushil — COMMENTED** [review]
Two spots outside this diff already do their own version of "strip query from hash path": `routes.utils.ts:44` (`path.split('?')[0]`) and `econsentSignature.ts` (`URLSearchParams`-based). After this change there are three independent implementations of the same operation with no shared helper — they can drift on edge cases (e.g. a literal `?` inside a query value) and reintroduce this bug class for a different param later. Worth consolidating into one helper used by all three call sites.

Also: `routerUtils.ts` has no test coverage before or after this change (`getCurrentPath`/`getLoginUrl` included), despite CONTRIBUTING.md and recent commit history (e.g. `b76f0fda3`) treating test coverage as enforced. The `getLoginUrl` regression noted inline shipped with nothing to catch it.

File: jsapp/js/router/routerUtils.ts (hunk 1 above)
Comment (umangzala-oc) [reply jsapp/js/router/routerUtils.ts]: Fixed. Added getCurrentRoutePath() which strips the query string, and switched isMyLibraryRoute() / isPublicCollectionsRoute() to use it. getCurrentPath() is reverted to return the full hash, so getLoginUrl() keeps econsent/event_type intact. Tests added. Verified locally.

**Review by gushil — APPROVED** [review]
Looks good to me. The regression from the earlier fix is resolved and covered by the new tests.

**Review by pkolli0408 — APPROVED** [review]
